Add STRVAL

rikardn opened the pull request:
Add `STRVAL`. Only for 38k for now since depends on `CHRGOT` and `FIN` that are 38k only.

FeralChild64 commented:
Unfortunately, there are conflicts. I might have merged changes in the wrong order...

rikardn commented:
The PRs where designed with no merger order in mind, but some conflicts arise regardless of order. I have rebased and fixed the conflicts. I also added a new similarity string that surfaced.
